### PR TITLE
CI checks for docs website (registry migration)

### PR DESCRIPTION
**Pull request by dak1n1**
### Description

* Remove `website` and `website-test` Makefile targets.
* Add new targets for linting website (and _test.go files).
* Add link-check for website.
* Add markdown check for website.
* Remove unused `errcheck` target.
  (We can use `tfproviderlint` for this when issue #887 is resolved).
* Fix all linting errors in website and _test.go files.

Since the provider documentation has been migrated to the Terraform Registry,
the Makefile targets `website` and `website-test` no longer serve a function,
and instead cause CI failures.

In place of these tests, the following checks have been added:
* terrafmt
* tfproviderdocs
* misspell (for common spelling errors. Not a full spell-check).
* markdownlint-cli
* markdown-link-check

All linting errors caught by these tools have also been fixed.

### Acceptance tests
- [x] Have you added an acceptance test for the functionality being added?
- [x] Have you run the acceptance tests on this branch?

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/master/CHANGELOG.md):

```release-note
Add linting for migrated website docs and linting for Terraform blocks within Go test files.
```

### References

https://github.com/hashicorp/terraform-provider-kubernetes/issues/953

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: website/docs/r/cluster_role.html.markdown [hunk 1]
```diff
@@ -57,19 +56,19 @@ The following arguments are supported:
...
 
 ## Nested Blocks
 
 ### `metadata`
 
 #### Arguments
 
-* `annotations` - (Optional) An unstructured key value map stored with the cluster role binding that may be used to store arbitrary metadata. 
+* `annotations` - (Optional) An unstructured key value map stored with the cluster role binding that may be used to store arbitrary metadata.
 **By default, the provider ignores any annotations whose key names end with *kubernetes.io*. This is necessary because such annotations can be mutated by server-side components and consequently cause a perpetual diff in the Terraform plan output. If you explicitly specify any such annotations in the configuration template then Terraform will consider these as normal resource attributes and manage them as expected (while still avoiding the perpetual diff problem).**
```
Comment (aareet) [inline website/docs/r/cluster_role.html.markdown:68]: @dak1n1 you may need to rebase this onto master to pick up the ~> changes I made to make these into Notes

**Comment by dak1n1** [comment]
Broken links on the website will probably have to be prioritized separately and handled in another PR. There are a lot of them.

https://gist.githubusercontent.com/dak1n1/e71930a082cbaaef4137a5cc8749697a/raw/7b3ca8a6311c5419ee905fa07bb83620c42a9bc8/gistfile1.txt

**Comment by dak1n1** [comment]
I just realized I never actually finished getting tfproviderdocs working. I was trying to set it up like the AWS provider did. I might handle that in a different PR if it looks like it'll be too time-consuming.

**Comment by aareet** [comment]
> Broken links on the website will probably have to be prioritized separately and handled in another PR. There are a lot of them.
> 
> https://gist.githubusercontent.com/dak1n1/e71930a082cbaaef4137a5cc8749697a/raw/7b3ca8a6311c5419ee905fa07bb83620c42a9bc8/gistfile1.txt

Created https://github.com/hashicorp/terraform-provider-kubernetes/issues/991

**Review by alexsomesan — APPROVED** [review]
LGTM. Quite useful stuff in here! nice!

**Comment by alexsomesan** [comment]
Looks like it isn't able to locate the `docker` binary in the Travis environment, although the Travis log clearly shows docker being present and started earlier. Not sure what's going on there. It may be that `command` is not available in there? Not sure. (you can check by switching to `which` instead)

**Comment by dak1n1** [comment]
Interestingly, Travis is able to use both `which` and `command` to find the docker binary.

Logs from Travis:
```
$ which docker
/usr/bin/docker
The command "which docker" exited with 0.
$ command -v docker
/usr/bin/docker
```

In this case, `which` *should* work better than `command`, since `which` is returning an exit code of 0, whereas `command` is not. However, they both still fail when `make` is run. So I'll just set the env variable to pass the path into `make` explicitly.

**Comment by dak1n1** [comment]
All make targets are passing now. (The only one I didn't test is `make testacc` since that already passed in a previous run).

https://gist.githubusercontent.com/dak1n1/f18e75515b16777d3ff2f4003d2808ed/raw/7ad4fb250e03e4d3fe6243f0da56f9574e7a78ca/gistfile1.txt

**Comment by dak1n1** [comment]
I finally got them all passing in Travis too. That was a lot different than running them locally.

**Comment by alexsomesan** [comment]
This looks great! thanks!

**Pull request merged**